src/commands/keygen.rs

```diff
@@ -0,0 +1,605 @@
+use std::ffi::OsString;
+use std::io::Write;
+use std::path::Path;
+
+use clap::{builder::ValueParser, Args, ValueEnum};
+use domain::base::iana::{DigestAlg, SecAlg};
+use domain::base::name::Name;
+use domain::base::zonefile_fmt::ZonefileFmt;
+use domain::sign::{common, GenerateParams};
+use domain::validate::Key;
+use lexopt::Arg;
+
+use crate::env::Env;
+use crate::error::{Context, Error};
+use crate::parse::parse_name;
+
+use super::{parse_os, parse_os_with, Command, LdnsCommand};
+
+#[derive(Clone, Debug, PartialEq, Eq, Args)]
+pub struct Keygen {
+    /// The signature algorithm to generate for
+    ///
+    /// Possible values:
+    /// - RSASHA256[:<bits>]: An RSA SHA-256 key (algorithm 8) of the given size (default 2048)
+    /// - ECDSAP256SHA256:    An ECDSA P-256 SHA-256 key (algorithm 13)
+    /// - ECDSAP384SHA384:    An ECDSA P-384 SHA-384 key (algorithm 14)
+    /// - ED25519:            An Ed25519 key (algorithm 15)
+    /// - ED448:              An Ed448 key (algorithm 16)
+    #[arg(
+        short = 'a',
+        long = "algorithm",
+        value_name = "ALGORITHM",
+        value_parser = ValueParser::new(Keygen::parse_algorithm),
+        verbatim_doc_comment,
+    )]
+    algorithm: GenerateParams,
+
+    /// Generate a key signing key instead of a zone signing key
+    #[arg(short = 'k')]
+    make_ksk: bool,
+
+    /// Whether to create symlinks.
+    #[arg(short, long, value_enum, num_args = 0..=1, require_equals = true, default_missing_value = "yes", default_value = "no")]
```

Comment:
Could you make the `short = 's'` and `long = "symlink"` explicit?

Reply:
This needs a massive comment to explain the rest of the attributes you put on here as well.